src/masternodes/rpc_icxorderbook.cpp

```diff
@@ -0,0 +1,1497 @@
+                        pcustomcsview->GetToken(order->idToken)->CreateSymbolKey(order->idToken), ScriptToString(offer->ownerAddress)));
+
+            if (submitdfchtlc.amount != offer->amount)
+                throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("cannot make dfc htlc with that amount, different amount necessary for offer (%s) - %s != %s!",
+                                offer->creationTx.GetHex(), ValueFromAmount(submitdfchtlc.amount).getValStr(), ValueFromAmount(offer->amount).getValStr()));
+
+            bool found = false;
+            pcustomcsview->ForEachICXSubmitDFCHTLCOpen([&found, &submitdfchtlc](CICXOrderView::TxidPairKey const & key, uint8_t i) {
+                if (key.first != submitdfchtlc.offerTx)
+                    return false;
+                found = true;
+                return false;
+            }, submitdfchtlc.offerTx);
+            if (found)
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "dfc htlc already submitted!");
+
+            std::unique_ptr<CICXSubmitEXTHTLCImplemetation> exthtlc;
+            pcustomcsview->ForEachICXSubmitEXTHTLCOpen([&](CICXOrderView::TxidPairKey const & key, uint8_t i) {
+                if (key.first == submitdfchtlc.offerTx)
+                {
+                    exthtlc = pcustomcsview->GetICXSubmitEXTHTLCByCreationTx(key.second);
+                    return (false);
+                }
+                return false;
+            }, submitdfchtlc.offerTx);
+            if (!exthtlc)
+                throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("offer (%s) needs to have ext htlc submitted first, but no external htlc found!",
+                            submitdfchtlc.offerTx.GetHex()));
+            if (submitdfchtlc.hash != exthtlc->hash)
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid hash, dfc htlc hash is different than extarnal htlc hash!");
+            if (submitdfchtlc.timeout < CICXSubmitDFCHTLC::MINIMUM_2ND_TIMEOUT)
+                throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Invalid parameters, argument \"timeout\" must be greater than %d", CICXSubmitEXTHTLC::MINIMUM_2ND_TIMEOUT - 1));
+            if (submitdfchtlc.timeout > (exthtlc->creationHeight + exthtlc->timeout) - targetHeight)
+                throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Invalid parameters, argument \"timeout\" must not be greater than expiration period of 1st htlc - %d", (exthtlc->creationHeight + exthtlc->timeout) - targetHeight));
+        }
+
+    }
+
+    CDataStream metadata(DfTxMarker, SER_NETWORK, PROTOCOL_VERSION);
+    metadata << static_cast<unsigned char>(CustomTxType::ICXSubmitDFCHTLC)
+             << submitdfchtlc;
+
+    CScript scriptMeta;
+    scriptMeta << OP_RETURN << ToByteVector(metadata);
+
+    const auto txVersion = GetTransactionVersion(targetHeight);
+    CMutableTransaction rawTx(txVersion);
+
+    CTransactionRef optAuthTx;
+    std::set<CScript> auths;
```

Comment:
Do we need this an the subsequent auths.size() check? If not can be removed, auths in arg to GetAuthInputsSmart can be {} and auths.size() in conditional below can be removed.